src/scenes/CollectionGalleryPage/CollectionGalleryHeader.tsx

```diff
@@ -0,0 +1,175 @@
+import { useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+import unescape from 'lodash.unescape';
+import { Subdisplay, BodyRegular } from 'components/core/Text/Text';
+import Spacer from 'components/core/Spacer/Spacer';
+import colors from 'components/core/colors';
+import Markdown from 'components/core/Markdown/Markdown';
+import NavElement from 'components/core/Page/GlobalNavbar/NavElement';
+import TextButton from 'components/core/Button/TextButton';
+import breakpoints, { size } from 'components/core/breakpoints';
+import CopyToClipboard from 'components/CopyToClipboard/CopyToClipboard';
+import { Collection } from 'types/Collection';
+import { useRouter } from 'next/router';
+import { useModal } from 'contexts/modal/ModalContext';
+import CollectionCreateOrEditForm from 'flows/shared/steps/OrganizeCollection/CollectionCreateOrEditForm';
+import noop from 'utils/noop';
+import Mixpanel from 'utils/mixpanel';
+import { usePossiblyAuthenticatedUser } from 'hooks/api/users/useUser';
+import MobileLayoutToggle from 'scenes/UserGalleryPage/MobileLayoutToggle';
+import { useBreakpoint } from 'hooks/useWindowSize';
+import { DisplayLayout } from 'components/core/enums';
+
+type Props = {
+  collection: Collection;
+  mobileLayout: DisplayLayout;
+  setMobileLayout: (mobileLayout: DisplayLayout) => void;
+};
+
+function CollectionGalleryHeader({ collection, mobileLayout, setMobileLayout }: Props) {
+  const { showModal } = useModal();
+  const { push } = useRouter();
+  const screenWidth = useBreakpoint();
+  const user = usePossiblyAuthenticatedUser();
+  const username = useMemo(() => window.location.pathname.split('/')[1], []);
```

Comment:
same comment here. though i'm pretty sure this logic already existed on another page!